src/vcpkg/registries.cpp

```diff
@@ -1151,9 +1168,18 @@ namespace vcpkg
     {
         if (data->second.stale)
         {
-            print2("Fetching registry information from ", data->first, "...\n");
-            data->second.value =
-                paths.git_fetch_from_remote_registry(data->first, "HEAD").value_or_exit(VCPKG_LINE_INFO);
+            StringView lockdata_key(data->first);
+            print2("Fetching registry information from ", lockdata_key, "...\n");
+
+            auto at_position = lockdata_key.end() - std::find(lockdata_key.begin(), lockdata_key.end(), '@');
```

Comment:
This does not work if the url has an `@` in it, such as `https://git@github.com`. There's no reason to inflict this kind of parsing pain on ourselves; we should store these strings separately.